src/acquisition/covid_hosp/common/database.py

```diff
@@ -39,15 +38,19 @@ def __init__(self,
     """
 
     self.connection = connection
-    self.table_name = table_name
-    self.hhs_dataset_id = hhs_dataset_id
-    self.publication_col_name = "issue" if table_name == 'covid_hosp_state_timeseries' or table_name == "covid_hosp_state_daily" else \
-      'publication_date'
-    self.columns_and_types = {
-      c.csv_name: c
-      for c in (columns_and_types if columns_and_types is not None else [])
-    }
-    self.key_columns = key_columns if key_columns is not None else []
+
+    if self.DATASET_NAME is not None: # populate from YAML file
+      self.table_name = chs.get_ds_table_name(self.DATASET_NAME)
+      self.hhs_dataset_id = chs.get_ds_dataset_id(self.DATASET_NAME)
+      self.publication_col_name = "issue" if self.table_name == 'covid_hosp_state_timeseries' or self.table_name == "covid_hosp_state_daily" else \
+        'publication_date'
```

Comment:
oof, that was a long comment that is better broken down into smaller parts.  lemme re-paste it with paragraph breaks and annotations:

**_[background/context on why different column names exist for the same purpose & why we shouldnt fix that now]_**

> hmmm... i wonder why this distinction exists at all, but im gonna guess that the different column name choices were made ~arbitrarily when these were first implemented. they could be be normalized to use only one, but its probably too late now since its been out in the wild for as long as it has.

**_[where to store differences in config]_**

> but if we are going to preserve individual cases for different datasets, it should be specified in the yaml definitions and not special-cased in this code here. i would also suggest assuming a default of "`issue`" if not specified, and let the `facility` dataset override with "`publication_date`".

**_[variable name change]_**

> in keeping with that theme (and to match the convention of the other sources in our purview), the column should be referred to in the code as "`issue_column`" instead of "`publication_col_name`".